plugins/restful/RestfulDataProviderVariable.php

```diff
@@ -0,0 +1,306 @@
+        if ($property_name == 'name') {
+          uksort($variables, array(__CLASS__, 'nameSort'));
+        }
+        elseif ($property_name == 'value') {
+          uasort($variables, array(__CLASS__, 'valueSort'));
+        }
+      }
+      if ($direction == 'DESC') {
+        $variables = array_reverse($variables, TRUE);
+      }
+    }
+
+    return $variables;
+  }
+
+  /**
+   * Set correct page for the index within the array of variables.
+   *
+   * Determine the page that should be seen. Page 1 is actually index 0.
+   *
+   * @param array $variables
+   *   An array keyed by variable name, valued by unserialized variable value.
+   *
+   * @throws \RestfulBadRequestException
+   */
+  protected function applyListPagination(array &$variables) {
+    list($offset, $range) = $this->parseRequestForListPagination();
+    $variables = array_slice($variables, $offset, $range);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getTotalCount() {
+    return array('restful_count' => count($this->getVariablesForList()));
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function index() {
+    $variables = $this->getVariablesForList();
+    $return = array();
+    foreach ($variables as $variable => $value) {
+      $return[] = $this->view($variable);
+    }
+
+    return $return;
+  }
+
+  /**
+   * Helper function to feed an array into viewMultiple().
+   *
+   * @param string $name_string
+   *  A string of variable names, separated by commas.
+   */
+  public function viewVariables($name_string) {
+    $names = array_unique(array_filter(explode(',', $name_string)));
+    return $this->viewMultiple($names);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewMultiple(array $names) {
+    if (empty($names)) {
+      return array();
+    }
+    $variables = array_keys($this->getVariablesForList());
+    $output = array();
+
+    // Build output according to filtered/sorted list of variables.
+    foreach ($variables as $variable) {
+      if (in_array($variable, $names)) {
+        $output[] = $this->view($variable);
+      }
+    }
+
+    return $output;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function view($name) {
+    $cache_id = array(
+      'tb' => 'variable',
+      'id' => $name,
+    );
+    $cached_data = $this->getRenderedCache($cache_id);
+    if (!empty($cached_data->data)) {
+      return $cached_data->data;
+    }
+
+    $return = array();
+    foreach ($this->getPublicFields() as $public_field_name => $info) {
+      if ($info['property'] == 'name') {
+        $public_field_value = $name;
+      }
+      elseif ($info['property'] == 'value') {
+        $public_field_value = variable_get($name);
+      }
+      else {
+        continue;
+      }
+
+      // Modify the public field value using a callback, if supplied.
+      $public_field_value = $info['callback'] ? static::executeCallback($info['callback'], array($value)) : $public_field_value;
```

Comment:
The `$info['callback']` provides an alternative to wire a callback to your public field. This should be something like:

```php
      if ($info['callback']) {
        $public_field_value = static::executeCallback($info['callback'], array($name));
      }
```